servicex/query_cache.py

```diff
@@ -70,7 +70,8 @@ def transformed_results(self, transform: TransformRequest,
 
     def cache_transform(self, record: TransformedResults):
         with self.lock:
-            self.db.insert(json.loads(record.model_dump_json()))
+            if not self.contains_hash(record.hash):
+                self.db.insert(json.loads(record.model_dump_json()))
```

Comment:
Just want to make sure - is there a race condition here? That is- do you need to `lock` the DB to make this query so no one inserts something between line 73 and 74?

Reply:
I'm confused - this _is_ already inside a lock, no? (`self.lock`)

Reply:
Ah - I thougth that was to prevent multi-threaded inside the single app, I was worried we were running multiple servicex app's.

Reply:
And this is the front end - not the ackend sign. Right - this looks great.

Reply:
Also it's a file lock, so it should actually work across multiple processes. (Which actually sets us up for an interesting race condition on its own ... @ketan96-m , does this only kick in once the submission to the backend is complete? Or is it added to the db before submission?)

Reply:
It is kicked in after the backend is complete